opentelemetry/proto/metrics/v1/metrics.proto

```diff
@@ -431,6 +441,134 @@ message HistogramDataPoint {
+  // must be zero. This value must be equal to the sum of the "sum" fields in
+  // buckets if a histogram is provided.
+  //
+  // Note: Sum should only be filled out when measuring non-negative discrete
+  // events, and is assumed to be monotonic over the values of these events.
+  // Negative events *can* be recorded, but sum should not be filled out when
+  // doing so.  This is specifically to enforce compatibility w/ OpenMetrics,
+  // see: https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#histogram
+  double sum = 5;
+
+  // scale describes the resolution of the histogram.  Boundaries are
+  // located at powers of the base, where:
+  //
+  //   base = (2^(2^-scale))
+  //
+  // The histogram bucket identified by `index`, a signed integer,
+  // contains values that are less than or equal to (base^index) and
+  // greater than (base^(index-1)).
+  //
+  // The positive and negative ranges of the histogram are expressed
+  // separately but both use the same scale to locate their boundaries.
+  //
+  // scale is valid in the range -4 through +8 inclusive.
+  sint32 scale = 6;
+
+  // zero_count is the count of values that are either exactly zero or
+  // within the region considered zero by the instrumentation at the
+  // tolerated degree of precision.  This bucket stores values that
+  // cannot be expressed using the standard exponential formula as
+  // well as values that have been rounded to zero.  Users have the
+  // option to set zero_tolerance to convey additional information about
+  // the width of the zero region.
+  //
+  // Implementations MAY consider the zero bucket to have probability
+  // mass equal to (zero_count / sum).
+  fixed64 zero_count = 7;
+
+  // zero_tolerance may be optionally set to convey the width of the
+  // zero region.  When this is not set, implementations may assume
+  // the zero tolerance equals the smallest representable positive
+  // IEEE 754 double-precision floating point number, which is (2^−1074).
+  //
+  // Producers can be configured to round measurements to zero that fall
+  // below the zero tolerance.  Users should avoid producing measurements
+  // that have meaningless precision near zero, as exponential buckets
+  // become increasingly dense in this region.
+  //
+  // Implementations MAY consider the zero bucket to have probability
+  // density equal to (zero_count / 2*zero_tolerance), if set.
+  double zero_tolerance = 8;
+
+  // positive carries the positive range of exponential bucket counts.
+  SparseBuckets positive = 9;
+
+  // negative carries the negative range of exponential bucket counts.
+  SparseBuckets negative = 10;
+
+  // SparseBuckets are a sparse set of bucket counts, compactly encoded.
+  message SparseBuckets {
+    // Span encodes a run of contiguous histogram buckets.
+    message Span {
+      sint32 offset = 1; // Gap to previous span, or starting point for 1st span (which can be negative).
```

Comment:
Same comment as Yuke made below about varint, it would be worth calling out this is a delta encoding for compression.

Reply:
I've accepted @yzhuge's larger point that allowing multiple Spans is a lot of added complexity for a transport protocol. I reworked the protocol to be simpler, but called out two places where we explicitly choose a varint encoding in the protocol, because of this concern.